cpp/include/cudf/ast/detail/expression_evaluator.cuh

```diff
@@ -277,48 +282,46 @@ struct expression_evaluator {
-    if (ref_type == detail::device_data_reference_type::COLUMN) {
+    if (input_reference.reference_type == detail::device_data_reference_type::COLUMN) {
       // If we have nullable data, return an empty nullable type with no value if the data is null.
-      auto const& table =
-        (device_data_reference.table_source == table_reference::LEFT) ? left : right;
+      auto const& table = (input_reference.table_source == table_reference::LEFT) ? left : right;
       // Note that the code below assumes that a right index has been passed in
-      // any case where device_data_reference.table_source == table_reference::RIGHT.
+      // any case where input_reference.table_source == table_reference::RIGHT.
       // Otherwise, behavior is undefined.
-      auto const row_index = (device_data_reference.table_source == table_reference::LEFT)
-                               ? left_row_index
-                               : *right_row_index;
+      auto const row_index =
+        (input_reference.table_source == table_reference::LEFT) ? left_row_index : *right_row_index;
       if constexpr (has_nulls) {
-        return table.column(data_index).is_valid(row_index)
-                 ? ReturnType(table.column(data_index).element<Element>(row_index))
+        return table.column(input_reference.data_index).is_valid(row_index)
+                 ? ReturnType(table.column(input_reference.data_index).element<Element>(row_index))
                  : ReturnType();
 
       } else {
-        return ReturnType(table.column(data_index).element<Element>(row_index));
+        return ReturnType(table.column(input_reference.data_index).element<Element>(row_index));
       }
-    } else if (ref_type == detail::device_data_reference_type::LITERAL) {
+    } else if (input_reference.reference_type == detail::device_data_reference_type::LITERAL) {
       if constexpr (has_nulls) {
-        return plan.literals[data_index].is_valid()
-                 ? ReturnType(plan.literals[data_index].value<Element>())
+        return plan.literals[input_reference.data_index].is_valid()
+                 ? ReturnType(plan.literals[input_reference.data_index].value<Element>())
                  : ReturnType();
 
       } else {
-        return ReturnType(plan.literals[data_index].value<Element>());
+        return ReturnType(plan.literals[input_reference.data_index].value<Element>());
       }
     } else {  // Assumes ref_type == detail::device_data_reference_type::INTERMEDIATE
       // Using memcpy instead of reinterpret_cast<Element*> for safe type aliasing
       // Using a temporary variable ensures that the compiler knows the result is aligned
-      IntermediateDataType<has_nulls> intermediate = thread_intermediate_storage[data_index];
+      IntermediateDataType<has_nulls> intermediate =
+        thread_intermediate_storage[input_reference.data_index];
       ReturnType tmp;
       memcpy(&tmp, &intermediate, sizeof(ReturnType));
```

Comment:
```suggestion
      std::memcpy(&tmp, &intermediate, sizeof(ReturnType));
```

Reply:
This is one case where we probably should not use the `std::` version.

Reply:
Agreed. Will there be a `cuda::std::memcpy` at some point that will dispatch to the appropriate CUDA function at some point, or is that simple enough that it'll never be a priority?

Reply:
I investigated offline and discussed with @vyasr. I saw some other `.cuh` / `.cu` files using `std::memcpy`, so my initial thought was that it seemed like it would be better C++ style. Although the two [should always compile identically in practice](https://godbolt.org/z/sezW4dYYd), that's only true if host C++ standard libraries define `std::memcpy` as an alias for `memcpy`. That could depend on the host's C++ standard library. To eliminate that possible issue, it seems best to rely on the C function `memcpy` instead. Resolving this conversation.